### PR TITLE
OKTA-642786 : G3 : Add nickname support for g3 and re-enable spec

### PR DESCRIPTION
**Pull request by glenfannin-okta**
## Description:

The purpose of this PR is to add nickname feature support for SIW G3 and enable the spec test to verify both versions.

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [x] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [x] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [x] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-642786](https://oktainc.atlassian.net/browse/OKTA-642786)

### Screenshot/Video:

Verification-data remediation: 
image

challenge authenticator remediation:
image

Select authenticator remediation:
image

Select authenticator remediation with long nickname:
image

### CODE REVIEW

File: src/v3/src/util/formUtils.ts
Comment (shawyu-okta) [inline src/v3/src/util/formUtils.ts:410]:
```suggestion
  const i18nMap: Record<IDX_STEP.CHALLENGE_AUTHENTICATOR | IDX_STEP.AUTHENTICATOR_VERIFICATION_DATA, Record<'sms' | 'voice', Record<'withPhoneWithNickName' | 'withPhoneWithoutNickName' | 'withoutPhone', string>>> = {
```
some formatting or defining some local union types might make it cleaner but you get the idea

File: src/v3/src/util/formUtils.ts
Comment (shawyu-okta) [inline src/v3/src/util/formUtils.ts:468]: caution: the `as string` cast is masking the fact that this can be `undefined`

File: src/v3/src/util/formUtils.ts
Comment (shawyu-okta) [inline src/v3/src/util/formUtils.ts:470]: Same here about the cast

File: src/v3/src/transformer/phone/phoneVerificationTransformer.ts
Comment (shawyu-okta) [inline src/v3/src/transformer/phone/phoneVerificationTransformer.ts:63]: let's not cast when possible - either use a guard to assure the TS compiler that this will be defined or allow the possibility of `undefined`

File: src/v3/src/components/AuthenticatorButton/AuthenticatorButton.tsx
Comment (shawyu-okta) [inline src/v3/src/components/AuthenticatorButton/AuthenticatorButton.tsx:202]: Can you attach screenshots of how this looks? Would be helpful to get some idea of what all these styles add up to. Additionally, might be good to update the UI Demo mock to show nickname in one of the authenticator buttons.

Comment (glenfannin-okta) [reply src/v3/src/components/AuthenticatorButton/AuthenticatorButton.tsx:202]: Attached screenshots, however, can you give more info re: the ui demo mock? I have seen it in some screenshots but not sure how you guys get that to display? Do you know what mock is used and do you just update the responseConfig file?

Comment (trevoring-okta) [reply src/v3/src/components/AuthenticatorButton/AuthenticatorButton.tsx:202]: I think he's talking about `_ui-demo.json` which you can add to the `responseConfig`
See: https://okta.slack.com/archives/C04NYNLLRDL/p1698082416073859

Comment (glenfannin-okta) [reply src/v3/src/components/AuthenticatorButton/AuthenticatorButton.tsx:202]: Ohh, awesome! Thank you! I will get that updated 🙏

File: src/v3/src/transformer/phone/phoneChallengeTransformer.ts
Comment (shawyu-okta) [inline src/v3/src/transformer/phone/phoneChallengeTransformer.ts:88]: consider inlining the logic above down here so you don't have to check for `undefined`

File: src/v3/src/util/idxUtils.ts
Comment (glenfannin-okta) [reply src/v3/src/util/idxUtils.ts:429]: Unfortunately not, [this](https://github.com/okta/okta-auth-js/blob/master/lib/idx/types/idx-js.ts#L99) is how it is defined in auth-js :(

File: src/v3/src/util/formUtils.ts
Comment (shawyu-okta) [inline src/v3/src/util/formUtils.ts:41]: Any way we can take advantage of this type in my other comment about the existence of a union type?

File: src/v3/src/util/formUtils.ts
Comment (shawyu-okta) [inline src/v3/src/util/formUtils.ts:443]: Before merging: are all of these translations already translated and merged into current `master`?

Comment (glenfannin-okta) [reply src/v3/src/util/formUtils.ts:443]: Yessir! Those were merged apart of [this](https://oktainc.atlassian.net/browse/OKTA-660939) ticket. Will wait to merge this until after this upcoming monthly to ensure the translations will be available when this is released.

File: src/v3/src/util/formUtils.ts
Comment (glenfannin-okta) [reply src/v3/src/util/formUtils.ts:41]: Oh, I can use it in the assertion part of that function, but for the parameter, I don't think I can use it in the type there unless I cast the parameter value before passing (which defeats the purpose of the guard). Unless you were thinking of something else?

File: src/v3/src/util/idxUtils.ts
Comment (glenfannin-okta) [reply src/v3/src/util/idxUtils.ts:429]: Ohh, you mean just `'sms' | 'voice'` this, sorry, I thought you were referring to `string | { form: IdxForm; } | Input[]` this 🤦‍♂️

**Review by shawyu-okta — APPROVED** [review]
Discussed directly - this can be merged into the current monthly release as all translations are already in `master`

File: src/v3/src/components/AuthenticatorButton/AuthenticatorButton.tsx
Comment (glenfannin-okta) [inline src/v3/src/components/AuthenticatorButton/AuthenticatorButton.tsx:196]: Since this is used as the describedBy of the button, this will get read aloud automatically regardless of the text length (if the ellipsis is displayed or not). 

We could addd a title property to the element for sighted user to allow them to hover to see the full text if it is truncated.


cc: @lesterchoi-okta as I think this is what you were going to comment on?

**Pull request merged**
